### DISCUSSION TITLE
How to train a regressor with 3D X_train?

### DISCUSSION THREAD

**Discussion by wnanao** [alan-turing-institute/sktime#6295] [Q&A]
Why does the process start training another model after all my epochs are trained? Is it because some of my parameters are not set correctly? In my understanding, the model training phase is generally over after all epochs have been trained, but the output area prints out the structural information and training information of model{n} and starts training another model. What part of the knowledge do I need to know or how to modify the code?
The following is my model definition code and output information：

```python
regressor = ResNetRegressor(n_epochs=1500, batch_size=128, verbose=True)
# regressor = InceptionTimeRegressor(n_epochs=2, batch_size=128, verbose=True)
regressor.fit(X_nor, y_nor)
```

![erro1](https://github.com/sktime/sktime/assets/70358014/c3fcbaff-37ee-490e-9f9b-420638653d52)

I found that the problem occurs in my input data, this does not occur when my training data X_train and y_train are 2D and 1D respectively. But when my X_train and y_trian are three-dimensional and two-dimensional respectively, like [n_instances, n_dimensions, series_length] and [n_instances, series_length]. The fit() function will train models of series_length length.
For example, the following code will train five times, 20 epochs each time:

```python
    X_train = np.random.rand(100, 12, 5)
    y_train = np.random.rand(100, 5)
    clf = ResNetRegressor(n_epochs=20, verbose=True)
    print("shape of X_train:", X_train.shape)
    print("shape of y_train:", y_train.shape)
    clf.fit(X_train, y_train)
```

![1713170858280](https://github.com/sktime/sktime/assets/70358014/561a88a2-4ad9-48c3-9316-aecc177af306)

**Comment by wnanao** [comment]
@fkiraly  Can you give me some helps?

**Comment by fkiraly** [comment]
Imo this is expected behaviour: if I look into the estimator, `ResNetRegressor` cannot deal with multivariate output `y` (in your case, `y_train`), because the `capability:multioutput` tag is `False`.

So, what the estimator will instead do, is train one instance of the model per column of `y`. In your case that's 5 models time 20 epochs, so 100 epochs overall.

Now, given the neural network architecture under the hood of `ResNetRegressor`, there is no inherent restriction of adapting it for multioutput `y` (you'd need to take care of the output layer being formatted right), it's just that it hasn't been done by the implementers, so it trains multiple models by default.

If you would like to give it a go, you'd be very welcome to contribute this extension!
